### PR TITLE
Enable the watch script to download and use the pre-built project manager.

### PR DESCRIPTION
**Pull request by MichaelMauderer**
### Pull Request Description
Adds functionality to download and start the project manager to the watch script.
Also add a command line toggle to enable/disable this behaviour.

### Checklist
Please include the following checklist in your PR:

- [x] The documentation has been updated if necessary.
- [x] All code conforms to the [Rust](https://github.com/enso-org/enso/blob/main/docs/style-guide/rust.md) style guide.
- [x] All code has automatic tests where possible.
- [x] All code has been profiled where possible.
- [x] All code has been manually tested in the IDE.
- [ ] ~All code has been manually tested in the "debug/interface" scene.~
- [x] All code has been manually tested by the PR owner against our [test scenarios](https://docs.google.com/spreadsheets/d/1RatJDM_f9_3bvYhl3Bpq2d8SyKgtVdrV1RkGxPU17c8/edit?ts=5faa7049#gid=0).
- [ ] All code has been manually tested by at least one reviewer against our [test scenarios](https://docs.google.com/spreadsheets/d/1RatJDM_f9_3bvYhl3Bpq2d8SyKgtVdrV1RkGxPU17c8/edit?ts=5faa7049#gid=0).

### CODE REVIEW

File: src/js/lib/client/src/preload.js [hunk 1]
```diff
@@ -1,4 +1,14 @@
...
+
+let win;
+if (remote !== undefined){
+    win = remote.getCurrentWindow()
+}
+
+if (win === undefined) {
+    console.warn("Could not get current window object for animation during preload.")
+}
+
```
Comment (wdanilo) [inline src/js/lib/client/src/preload.js:11]: The error should be more specific - instead fo "animation during preload" it should tell "window startup animation"

File: src/js/lib/client/src/preload.js [hunk 2]
```diff
@@ -10,6 +20,9 @@ function ease_in_out_quad(t) {
 }
 
 function animate_show(target) {
+    if (target === undefined) {
+        return
+    }
```
Comment (wdanilo) [inline src/js/lib/client/src/preload.js:25]: I dont think this is the respoinsibility of this function to check if the arg was correct. Instead, check it on the call side, please.

**Review by wdanilo — APPROVED** [review]

**Review by wdanilo — CHANGES_REQUESTED** [review]
Why these are not checked?
image

**Comment by MichaelMauderer** [comment]
> Why these are not checked?
> image

`debug/interface` is not applicable, and the other box has been done now.

File: build/run.js [hunk 3]
```diff
@@ -72,18 +72,28 @@ function command(docs) {
...
-//    child_process.execFile(bin_path, [], (error, stdout, stderr) => {
-//        console.error(stderr)
-//        if (error) {
-//            throw error
-//        }
-//        console.log(stdout)
-//    })
-//}
+/// Build the project manager module, which downloads the project manager binary for the current
+// platform.
```
Comment (wdanilo) [inline build/run.js:76]: use 3 `/` instead of 2

**Review by wdanilo — APPROVED** [review]

**Pull request merged**